test/vinyl-luatest/update_optimize_test.lua

```diff
@@ -0,0 +1,405 @@
+local t = require('luatest')
+local g = t.group()
+local fiber = require('fiber')
+local log = require('log')
+
+local space
+local dump_count
+local new_stmt_count
+local old_stmt_count
+local index, index2, index3
```

Comment:
These shouldn't be global variables. Please move them to every test case that uses them. I think you should create a new space in each test case and drop it at the end. The original test is quite messy in that regard - instead of re-creating a space each time it tries to reuse it (e.g. by truncating it). AFAIU the only reason for that is to save some typing and that shouldn't be done in the first place.

Reply:
Fixed